src/util/address.rs

```diff
@@ -479,28 +480,40 @@ use crate::vm::VMBinding;
 /// the opaque `ObjectReference` type, and we haven't seen a use case for now.
 #[repr(transparent)]
 #[derive(Copy, Clone, Eq, Hash, PartialOrd, Ord, PartialEq, NoUninit)]
-pub struct ObjectReference(usize);
+pub struct ObjectReference(NonZeroUsize);
```

Comment:
The comments for the type should note that `ObjectReference` cannot be null, and for places where we may have an invalid object reference, we use `Option<ObjectReference>`. `NonZeroUsize` makes sure that `Option<ObjectReference>` has the same length as `ObjectReference`.